src/accel/TrackingManagerIntegration.cc

```diff
@@ -0,0 +1,136 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/TrackingManagerIntegration.cc
+//---------------------------------------------------------------------------//
+#include "TrackingManagerIntegration.hh"
+
+#include <memory>
+#include <G4ParticleDefinition.hh>
+#include <G4Run.hh>
+#include <G4Threading.hh>
+
+#include "geocel/GeantUtils.hh"
+
+#include "TrackingManager.hh"
+
+#include "detail/IntegrationSingleton.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Access the public-facing integration singleton.
+ */
+TrackingManagerIntegration& TrackingManagerIntegration::Instance()
+{
+    static TrackingManagerIntegration tmi;
+    return tmi;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Edit options before starting the run.
+ */
+SetupOptions& TrackingManagerIntegration::Options()
+{
+    return detail::IntegrationSingleton::instance().setup_options();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize during ActionInitialization on non-worker thread in MT mode.
+ */
+void TrackingManagerIntegration::BuildForMaster()
+{
+    CELER_VALIDATE(
+        G4Threading::IsMasterThread()
+            || G4Threading::IsMultithreadedApplication(),
+        << R"(BuildForMaster called from a worker thread or non-MT code)");
+
+    detail::IntegrationSingleton::instance().initialize_logger();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize during ActionInitialization.
+ */
+void TrackingManagerIntegration::Build()
+{
+    if (G4Threading::IsMasterThread())
+    {
+        CELER_VALIDATE(!G4Threading::IsMultithreadedApplication(),
+                       << "cannot call Integration::Build from worker thread "
+                          "in a multithreaded application");
```

Comment:
Is this to protect against a user calling this from `ActionInitialization::BuildForMaster()`?

Maybe we could add a comment on why this is only called in serial mode (logger should only be initialized by the master thread?).